example/fine_advection/advection_package.cpp

```diff
@@ -0,0 +1,180 @@
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <coordinates/coordinates.hpp>
+#include <globals.hpp>
+#include <parthenon/package.hpp>
+
+#include "advection_package.hpp"
+#include "defs.hpp"
+#include "kokkos_abstraction.hpp"
+#include "reconstruct/dc_inline.hpp"
+#include "utils/error_checking.hpp"
+
+using namespace parthenon::package::prelude;
+
+// *************************************************//
+// define the "physics" package Advect, which      *//
+// includes defining various functions that control*//
+// how parthenon functions and any tasks needed to *//
+// implement the "physics"                         *//
+// *************************************************//
+
+namespace advection_package {
+using parthenon::UserHistoryOperation;
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto pkg = std::make_shared<StateDescriptor>("advection_package");
+
+  Real cfl = pin->GetOrAddReal("Advection", "cfl", 0.45);
+  pkg->AddParam<>("cfl", cfl);
+
+  Real vx = pin->GetOrAddReal("Advection", "vx", 1.0);
+  Real vy = pin->GetOrAddReal("Advection", "vy", 0.0);
+  Real vz = pin->GetOrAddReal("Advection", "vz", 0.0);
+  pkg->AddParam<>("vx", vx);
+  pkg->AddParam<>("vy", vy);
+  pkg->AddParam<>("vz", vz);
+
+  Real refine_tol = pin->GetOrAddReal("Advection", "refine_tol", 0.3);
+  pkg->AddParam<>("refine_tol", refine_tol);
+  Real derefine_tol = pin->GetOrAddReal("Advection", "derefine_tol", 0.03);
+  pkg->AddParam<>("derefine_tol", derefine_tol);
+
+  auto profile_str = pin->GetOrAddString("Advection", "profile", "wave");
+  if (!((profile_str == "wave") || (profile_str == "smooth_gaussian") ||
+        (profile_str == "hard_sphere") || (profile_str == "block"))) {
+    PARTHENON_FAIL(("Unknown profile in advection example: " + profile_str).c_str());
+  }
+  pkg->AddParam<>("profile", profile_str);
+
+  pkg->AddField<Conserved::scalar_fine>(
+      Metadata({Metadata::Cell, Metadata::Fine, Metadata::Independent,
+                Metadata::WithFluxes, Metadata::FillGhost}));
+
+  pkg->AddField<Conserved::scalar>(Metadata({Metadata::Cell, Metadata::Independent,
+                                             Metadata::WithFluxes, Metadata::FillGhost}));
+  pkg->AddField<Conserved::scalar_fine_restricted>(
+      Metadata({Metadata::Cell, Metadata::Derived, Metadata::OneCopy}));
+  pkg->CheckRefinementBlock = CheckRefinement;
+  pkg->EstimateTimestepMesh = EstimateTimestep;
+  pkg->FillDerivedMesh = RestrictScalarFine;
```

Comment:
if we want to eventually replace sparse advection driver should one of these be a sparse pool? Or were you thinking of replacing dense advection? Maybe it should all be unified in a single new advection example...

Reply:
When I replace the sparse advection test I was going to add some sparse variables here. I also would need to update all the loops to use hierarchical parallelism since they are now written to go over the number of variables in the pack on the first block. This would be a small change though. 

I think it is probably better not to remove the original advection example, since it uses some history stuff and has a number of tests that rely on it. Maybe it should be moved from the example directory to the tst directory though.

Reply:
I also think we may want to add the vector wave equation to this test.

Reply:
I like the idea of moving the old advection example to the test directory.